pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2382,6 +2460,35 @@ protected void internalUpdateSubscriptionProperties(AsyncResponse asyncResponse,
         });
     }
 
+    protected void internalAnalyzeSubscriptionBacklog(AsyncResponse asyncResponse, String subName,
+                                                      Optional<Position> position,
+                                                      boolean authoritative) {
+        CompletableFuture<Void> future;
+        if (topicName.isGlobal()) {
+            future = validateGlobalNamespaceOwnershipAsync(namespaceName);
+        } else {
+            future = CompletableFuture.completedFuture(null);
+        }
+
+        future.thenCompose(__ -> validateTopicOwnershipAsync(topicName, authoritative)).thenAccept(__ -> {
+            if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName,
+                    authoritative, false).partitions > 0) {
+                throw new RestException(Status.METHOD_NOT_ALLOWED,
+                        "Analyze backlog on a partitioned topic is not allowed, "
+                                + "please try do it on specific topic partition");
+            }
```

Comment:
We should check the topic is a partitioned topic or not first? If it's a partitioned topic, the ownership checking is redundant

Reply:
other methods check if the topic is partitioned in this same place.
if the broker is not the owner we will be redirected to another broker and we won't reach this point.

I notice that getPartitionedTopicMetadata is sync, I have reworked this part to make it fully async